### ISSUE TITLE
Too much quoting of ppdOptions

### ISSUE THREAD

**Issue by jceb** [NixOS/nixpkgs#317931]
### Describe the bug

When I set ppdOption according to the documentation, the default options aren't properly set in cups.

### Steps To Reproduce

I tried to set ppdOptions for my printer as follows:

```nix
  hardware.printers.ensurePrinters = [
    {
      name = "samsung_m288x";
      location = "Home";
      model = "foomatic-db-ppds/Samsung-M288x-pxlmono.ppd.gz";
      deviceUri = "dnssd://Samsung%20M288x%20Series%20(SEC84251910B3C9)._printer._tcp.local/";
      ppdOptions = {
        Duplex = "DuplexNoTumble";
        PageSize = "A4";
      };
    }
  ];
```

This produces the following ensure-printers start script:

```bash
#!/nix/store/306znyj77fv49kwnkpxmb0j2znqpa8bj-bash-5.2p26/bin/bash
set -e
/nix/store/b3f8bs5njwk6ccihfn05wrpjz6xg9czm-cups-2.4.7/bin/lpadmin '-L' 'Home' '-m' 'foomatic-db-ppds/Samsung-M288x-pxlmono.ppd.gz' '-o' ''\''Duplex'\''='\''DuplexNoTumble'\''' '-o' ''\''PageSize'\''='\''A4'\''' '-p' 'samsung_m288x' '-v' 'dnssd://Samsung%20M288x%20Series%20(SEC84251910B3C9)._printer._tcp.local/' -E

systemctl stop cups.service
```

Executing the script succeeds, however the default options aren't set properly:

```bash
lpoptions -l
PrintoutMode/Print Quality: Draft *Normal High
PageSize/Page Size: Custom.WIDTHxHEIGHT *Letter A4 11x17 A3 A5 B5 Env10 EnvC5 EnvDL EnvISOB5 EnvMonarch Executive Legal
InputSlot/Media Source: *Default Upper Middle Lower Multipurpose Automatic Manual
Duplex/Double-Sided Printing: DuplexNoTumble DuplexTumble *None
PrinterResolution/Resolution: *FromPrintoutMode 300x300dpi 600x600dpi 1200x600dpi 1200x1200dpi
```

When I manually remove all `'\''` quotes from the command, it properly sets the default options.

### Expected behavior

Default options as defined in the documentation are reflected in cups.

### Notify maintainers

@rhoriguchi @florianjacob 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.12, NixOS, 24.05 (Uakari), 24.05.803.b3b2b28c1daa`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"home-manager-24.05.tar.gz, home-manager-23.05-23.05.tar.gz, home-manager-23.11-23.11.tar.gz, nixos-24.05, nixos-23.05-23.05, nixos-23.11-23.11, nixos-unstable"`
 - channels(jceb): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

**Comment by rhoriguchi** [comment]
Should be fixed by #312261

**Comment by jceb** [comment]
Awesome, thank you!
